beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1299,9 +1305,116 @@ impl<T: BeaconChainTypes> ExecutionPendingBlock<T> {
         /*
          *  Perform cursory checks to see if the block is even worth processing.
          */
-
         check_block_relevancy(block.as_block(), block_root, chain)?;
 
+        // Spawn an async call to the execution node to fetch blobs.
+        let blob_chain = chain.clone();
+        let blob_block = block.as_block().clone();
+        let blob_fetcher_future = async move {
+            let block = blob_block;
+            let chain = blob_chain;
+
+            if !block.message().body().has_blobs() {
+                debug!(chain.log, "Blobs from EL - none required");
+                return Ok(());
+            }
+
+            let execution_layer = chain
+                .execution_layer
+                .as_ref()
+                .ok_or(BeaconChainError::ExecutionLayerMissing)?;
+            let execution_payload = block.message().execution_payload()?;
+            let Some(transactions) = execution_payload.transactions() else {
+                debug!(chain.log, "Blobs from EL - blinded payload");
+                return Ok(());
+            };
+            debug!(chain.log, "Blobs from EL - decoding");
+            let blob_ids =
+                extract_blob_transaction_ids::<T::EthSpec>(transactions).map_err(|e| {
+                    BlockError::ExecutionPayloadError(ExecutionPayloadError::RequestFailed(
+                        execution_layer::Error::VerifyingVersionedHashes(e),
+                    ))
+                })?;
+            let num_blob_tx = blob_ids.len();
+
+            debug!(
+                chain.log,
+                "Blobs from EL - start request";
+                "num_blob_tx" => num_blob_tx,
+            );
+            let response = execution_layer.get_blobs(blob_ids).await.map_err(|e| {
+                BlockError::ExecutionPayloadError(ExecutionPayloadError::RequestFailed(e))
+            })?;
+            let num_fetched_tx = response.blobs.iter().filter(|b| b.is_some()).count();
+            if num_fetched_tx == 0 {
+                debug!(chain.log, "Blobs from EL - response with none");
+                return Ok(());
+            } else if num_fetched_tx < num_blob_tx {
+                debug!(
+                    chain.log,
+                    "Blobs from EL - response with some";
+                    "fetched" => num_fetched_tx,
+                    "total" => num_blob_tx,
+                );
+            } else {
+                debug!(
+                    chain.log,
+                    "Blobs from EL - response with all";
+                    "num_blob_tx" => num_fetched_tx
+                );
+            }
+            let mut fixed_blob_sidecar_list = FixedBlobSidecarList::default();
+            for (i, (blob, kzg_proof)) in response
+                .blobs
+                .into_iter()
+                .flatten()
+                .flat_map(|blob| blob.blobs.into_iter().zip(blob.proofs))
+                .enumerate()
```

Comment:
I've just realised there's a bug here when the EL returns a subset of blobs. The indices get out of whack because we enumerate only the blobs that were returned, rather than leaving gaps for the ones that were not. Will fix.